src/file.cpp

```diff
@@ -278,6 +278,7 @@ namespace
 	}
 }
 # else
+#  undef _BSD_SOURCE
```

Comment:
is this safe? or is it better to only define it if it isn't already defined? I don't know, maybe it has some important value